architecture_docs/ADRs/0000-use_architectural_decision_records.md

```diff
@@ -0,0 +1,198 @@
+# Use Architectural Decision Records
+
+# Status
+
+Accepted
+
+## Context & Problem Statement
+
+There are some choices we made in the past where we debated, weighed alternatives and tradeoffs,
+but we didn't document our choices or why we chose them.
+This means that newcomers to the project might not undertand why these choices happened.
+But even longtime contributors can forget!
+
+The benefits of ADRs have been outlined in [Michael Nygard's "Documenting architecture decisions"][nygard-blog].
+There are additional references in [this Red Hat blog post about ADRs](https://www.redhat.com/architect/architecture-decision-records).
+
+Stylistically, there are many formats to consider, each with different tradeoffs. How meta!
+
+## Decision
+
+We will use ADRs. 
+
+ADRs must be for important decisions regarding Pelorus's architecture, not to pretend we are IETF members.
+
+Proposed ADRs will be written up in the below format, and opened as a Pull Request.
+
+### The Format
+
+We will use GitHub Flavored Markdown documents.
+
+The final file name will be a unique sequential number that follows the `NNNN-title_after_dash_with_underscores` pattern.  
+For example, `0000-use_architectural_decision_records.md`, which would be followed by `0001-some_other_adr.md`.  
+
+The filename title may be a shorter version of the document title,
+but they should try to remain consistent.
+
+The file content sections will be as below.
+The title will be an `h1` (`#`).
+
+All other sections will be `h2`s (`##`) with their names verbatim,
+as listed below before the colon.
+
+Subsubsections (`h3` / `###` and lower) may be applied
+as necessary, except for in the `Status` field.
+
+- Title: indicate the content.
+  - Notes to the reader may go below the title.
+    Subsections may not be used here.
+    These are for changing how one might interpret the below content (e.g. format changes).  
+    This should be rare.
+- Status: see [statuses](####statuses).
+- Context & Problem Statement: describe the current situation; why this is necessary.
+- Decision: the "what" and "how" of the decision.
+- Rationale: the "why" of the decision. Can also be thought of as the context for the _decision_ more than the problem.
+- Alternatives Considered: if applicable. Includes why they were not chosen.
+- For Future Discussion: if applicable. Includes the parts of the discussion that should be revisited later. If these are revisited in a future ADR, that ADR should be linked here.
+- Addendum: optional, additional, less-critical notes to the reader.
+- Consequences: As a retrospective, describe the effect of the decision, listing positive and negative aspects. This is for reflecting after the change, not predicting the future.
+
+
+#### Statuses
```

Comment:
I would remove the status (this would remove the undecided ones from being merged) and I would not care if they are obsoleted or not (because I believe we will forget to update them). I would see them as a changelog. At that time in history we decided to do something, but this does not mean we are still doing that